packages/pwa/app/utils/site-utils.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {getConfig, getParamsFromUrl} from './utils'
+import {JSONPath} from 'jsonpath-plus'
+/**
+ * This functions takes an url and returns a site object,
+ * an error will be thrown if not url is passed in or no site is found
+ * @param {string} url
+ * @returns {object}
+ */
+export const resolveSiteFromUrl = (url) => {
+    if (!url) {
+        throw new Error('url is required to find a site object.')
+    }
+    const {hostname} = new URL(url)
+
+    const sites = getSites()
+    const defaultSiteId = getConfig('app.defaultSiteId')
+    let site
+    // Step 1: look for the site from the url, if found, return the site
+    site = getSiteByUrl(url)
```

Comment:
The name here `getSiteByUrl` is a little misleading as in the `url` which is inclusive of the hostname is getting passed in, but not taken into account. If you want to be more accurate you might want to do call it `getSiteByPath` where path is inclusive of the pathname and the search and the hash. Like this: 

```
const {hostname} = new URL(url)
const path = `${pathname}${search}${hash}`

// Step 1: look for the site from the url, if found, return the site
site = getSiteByPath(path)

```